File: Minimum Increment to Make Array Unique/Minimum_increment.py
class Solution:
    def minIncrementForUnique(self, nums: List[int]) -> int:
        if not nums: 
            return 0
        nums.sort()
        res = 0
        for i in range(1, len(nums)):
            if nums[i] <= nums[i - 1]:
                incr = nums[i - 1] - nums[i] + 1
                res += incr
                nums[i] += incr
        return res

File: Minimum Increment to Make Array Unique/Minimum_increment.cpp
class Solution {
public:
    int minIncrementForUnique(std::vector<int>& nums) {
        if (nums.empty()) return 0;
        sort(nums.begin(), nums.end());
        int res = 0;
        for (size_t i = 1; i < nums.size(); ++i) {
            if (nums[i] <= nums[i - 1]) {
                res += (nums[i - 1] - nums[i] + 1);
                nums[i] = nums[i - 1] + 1;
            }
        }
        
        return res;
    }
};